File: ProgettoKalk/KalkMV/Model/figura.cpp
#include "figura.h"

unsigned int Figura::id=1;

Figura::Figura(string s):nome(s){
    id++;
}

string Figura::getNome() const {
    return nome;
}

void Figura::setNome(string s){
    nome=s;
}

unsigned int Figura::getId() const {
    return id;
}

File: ProgettoKalk/KalkJava/Figura.java
public abstract class Figura {
    private String nome;
    private static  int id=1;

    public Figura(String s){
        nome=s;
        id++;
    }
    public String getNome(){
        return nome;
    }
    public void setNome(String s){
        nome=s;
    }
    public int getId(){
        return id;
    }
    public abstract double distanza(Figura f);
    public abstract String toString();

}
